next-cmake/CMakeLists.txt

```
@@ -0,0 +1,367 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+# ######################################################################### Copyright
+
+cmake_minimum_required(VERSION 3.25.2)  # TODO: Test this with various cmake versions
+project(rocblas VERSION 4.5.0 LANGUAGES CXX)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+include(FetchContent)
+include(fetch-rocm-cmake)
+include(CMakeDependentOption)
+include(CMakePackageConfigHelpers)
+include(GenerateExportHeader)
+include(GNUInstallDirs)
+include(ROCMSetupVersion)
+include(ROCMClients)
+include(ROCMCreatePackage)
+include(ROCMInstallTargets)
+include(ROCMPackageConfigHelpers)
+
+if(POLICY CMP0135)
+  cmake_policy(SET CMP0135 NEW)
+endif()
+
+# Export compile_commands.json for clang tooling support
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+set(ROCBLAS_ENABLE_DEVICE ON CACHE BOOL "Build rocBLAS device libraries.")
+set(ROCBLAS_ENABLE_HOST ON CACHE BOOL "Build rocBLAS host library.")
+cmake_dependent_option(ROCBLAS_ENABLE_CLIENT "Build rocBLAS clients." ON ROCBLAS_ENABLE_HOST OFF)
+
+if(ROCBLAS_ENABLE_HOST)
+    set(ROCBLAS_BUILD_SHARED ON CACHE BOOL "Build the rocBLAS shared or static library")
+    set(ROCBLAS_ENABLE_MSGPACK ON CACHE BOOL "Enable msgpack support.")
+    set(ROCBLAS_ENABLE_OPENMP ON CACHE BOOL "Enable OpenMP support.")
+    set(ROCBLAS_ENABLE_BLIS ON CACHE BOOL "Enable BLIS support.") # I don't know that we can build with this OFF
+    set(ROCBLAS_ENABLE_LLVM OFF CACHE BOOL "Enable LLVM support.")
+    set(ROCBLAS_ENABLE_TENSILE ON CACHE BOOL "Build rocBLAS host library with Tensile backend.")
+    set(ROCBLAS_ENABLE_HIPBLASLT OFF CACHE BOOL "Build rocBLAS host library with hipBLASLt backend (not currently supported).")
+endif()
+
+if(ROCBLAS_ENABLE_CLIENT)
+    # rocm-smi is not presently available on Windows so we do not require it.
+    set(ROCBLAS_ENABLE_BENCHMARKS ON CACHE BOOL "Build benchmark client.")
+    set(ROCBLAS_ENABLE_TESTS ON CACHE BOOL "Build benchmark client.")
+    set(ROCBLAS_ENABLE_SAMPLES ON CACHE BOOL "Build client samples.")
+    set(ROCBLAS_ENABLE_FORTRAN OFF CACHE BOOL "Build Fortran clients.")
+    cmake_dependent_option(ROCBLAS_REQUIRE_ROCM_SMI "Require rocm_smi." ON "NOT WIN32" OFF)
+endif()
+
+
+find_package(hip REQUIRED)
+
+if(ROCBLAS_ENABLE_BLIS)
+    find_package(BLIS REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_OPENMP)
+    find_package(OpenMP REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_LLVM)
+    find_package(LLVM REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_HIPBLASLT)
+    find_package(roc::hipblaslt REQUIRED)
+    rocm_package_add_dependencies(DEPENDS "hipblaslt >= 0.6.0")
+endif()
+
+if(ROCBLAS_REQUIRE_ROCM_SMI)
+    find_package(rocm_smi REQUIRED)
+else()
+    find_package(rocm_smi)
+endif()
```

```suggestion
if(ROCBLAS_REQUIRE_ROCM_SMI)
    find_package(rocm_smi REQUIRED)
endif()
```

don't find it if it isn't required. Also shouldn't we check if we are on windows? See comment above. How is rocm-smi used in rocBLAS?

---

I have added in checks to evaluate whether the package is found, therefore, we can still request to find it if it's not required and conditionally link it into the `rocblas-clients-common` library.